projects/common/src/telemetry/telemetry.ts

```diff
@@ -0,0 +1,43 @@
+import { InjectionToken } from '@angular/core';
+import { Dictionary } from './../utilities/types/types';
+
+export interface UserTelemetry {
+  readonly config: UserTelemetryConfig;
+  readonly initialized: boolean;
+  initialize(): void;
+  identify(userTraits: UserTraits): void;
+  trackEvent(name: string, eventData: Dictionary<unknown>): void;
+  trackPage(url: string): void;
+  shutdown(): void;
+}
+
+export interface UserTraits extends Dictionary<unknown> {
+  email: string;
+  companyName: string;
+  name: string;
+  displayName: string;
+}
+
+export interface UserTelemetryConfig<TConfig extends TelemetryProviderConfig = TelemetryProviderConfig> {
+  telemetryProvider: UserTelemetryType | string;
+  telemetryProviderConfig: TConfig;
+  enablePageTracking: boolean;
+  enableEventTracking: boolean;
+  enableErrorTracking: boolean;
+}
+
+export interface TelemetryProviderConfig {
+  orgId: string;
+}
+
+export const enum UserTelemetryType {
```

Comment:
That makes this code more useful. I understand that pluggable model is good and I want to support that. But OOTB would make it easier for multiple consumers, in this case hypertrace and traceable.

Reply:
There's a difference between having support for certain ones out of the box vs having the common code know about them - that's what I was trying to get at here. Having OOTB support is fine, but if we take in injection tokens instead of well known strings, the common code doesn't need to know about any specific provider - which we'll need anyway to support providing custom ones.


So basically:
```ts
    UserTelemetryModule.withProviders([
      {
        telemetryProvider: FullStoryTelemetryProvider, // Instead of: UserTelemetryType.FullStory,
        telemetryProviderConfig: { orgId: environment.fullStoryOrgId },
        enablePageTracking: true,
        enableEventTracking: true,
        enableErrorTracking: true
      }
    ]);

Reply:
So what is provider in this case? Is it a token or a class implementation object?